api/db.py

```diff
@@ -68,6 +68,13 @@ async def find_by_attributes(self, model, attributes):
         data = await col.find(attributes).to_list(None)
         return list(model(**obj) for obj in data)
 
+    def get_model_from_kind(self, kind: str):
+        """Get model from kind parameter"""
+        for key, value in self.COLLECTIONS.items():
```

Comment:
Alternatively we could have a reversed dictionary created directly in the class, to avoid having to iterate through it every time. For example:
```python
    COLLECTIONS = {
        'foo': 'user',
        'bar': 'node',
        'baz': 'regression',
    }

    KINDS = {
        kind: collection
        for collection, kind in COLLECTIONS.items()
    }
```

Also, there's a slight problem here as the `COLLECTIONS` really are the name of the database collections. They may not always be the same as the models defined in `api.models`. So maybe this functionality should be implemented in `api.models` instead.

I think the whole model / collection / kind design needs to be reconsidered in the light of our current understanding of how this is meant to work in order to avoid potential issues or confusion in the semantics in the future.

Reply:
Added `get_model_from_kind` to `api.models`.